drafts/2019-05-16-restful-config-viewing-and-validation-service.adoc

```diff
@@ -0,0 +1,327 @@
+      "enableConnectionCasting": false,
+      "jdbcDriverRef": [
+         {
+            "configElementName": "jdbcDriver",
+            "uid": "dataSource[DefaultDataSource]/jdbcDriver[default-0]",
+            "libraryRef": [
+               {
+                  "configElementName": "library",
+                  "uid": "derby",
+                  "id": "derby",
+                  "apiTypeVisibility": "spec,ibm-api,api,stable",
+                  "fileRef": [
+                     {
+                        "configElementName": "file",
+                        "uid": "library[derby]/file[default-0]",
+                        "name": "/home/nmittles/git/liberty/open-liberty/dev/build.image/wlp/usr/servers/blog/derby/derby.jar"
+                     }
+                  ]
+               }
+            ]
+         }
+      ],
+      "statementCacheSize": 10,
+      "syncQueryTimeoutWithTransactionTimeout": false,
+      "transactional": true,
+      "properties.derby.embedded": [
+         {
+            "createDatabase": "create",
+            "databaseName": "memory:defaultdb",
+            "password": "******",
+            "user": "dbuser"
+         }
+      ],
+      "api": [
+         "/ibm/api/validation/dataSource/DefaultDataSource"
+      ]
+   }
+]
+```
+
+To view an individual data source, append the data source's `uid` as seen from viewing the configuration. In our case this is `DefaultDataSource`, and results in a URL of:
+https://localhost:9443/ibm/api/config/dataSource/DefaultDataSource.
+
+```json
+{
+   "configElementName": "dataSource",
+   "uid": "DefaultDataSource",
+   "id": "DefaultDataSource",
+   "beginTranForResultSetScrollingAPIs": true,
+   "beginTranForVendorAPIs": true,
+   "connectionSharing": "MatchOriginalRequest",
+   "enableConnectionCasting": false,
+   "jdbcDriverRef": [
+      {
+         "configElementName": "jdbcDriver",
+         "uid": "dataSource[DefaultDataSource]/jdbcDriver[default-0]",
+         "libraryRef": [
+            {
+               "configElementName": "library",
+               "uid": "derby",
+               "id": "derby",
+               "apiTypeVisibility": "spec,ibm-api,api,stable",
+               "fileRef": [
+                  {
+                     "configElementName": "file",
+                     "uid": "library[derby]/file[default-0]",
+                     "name": "/home/nmittles/git/liberty/open-liberty/dev/build.image/wlp/usr/servers/blog/derby/derby.jar"
+                  }
+               ]
+            }
+         ]
+      }
+   ],
+   "statementCacheSize": 10,
+   "syncQueryTimeoutWithTransactionTimeout": false,
+   "transactional": true,
+   "properties.derby.embedded": [
+      {
+         "createDatabase": "create",
+         "databaseName": "memory:defaultdb",
+         "password": "******",
+         "user": "dbuser"
+      }
+   ],
+   "api": [
+      "/ibm/api/validation/dataSource/DefaultDataSource"
+   ]
+}
+```
+
+== Validating a Data Source
```

Comment:
we should talk about DS validation before we talk about viewing config, since I expect readers will mainly care about DS validation and understanding the config API isn't a pre-req for understanding the validation API

Reply:
Moved "Validating a Data Source" and "Validating a JCA Connection Factory" before their respective config viewing sections.